File: mobile/src/api/brainDump.ts
import { apiRequest } from "./client";

export type BrainDumpSignals = {
  emotional_state: string | null;
  blockers: string[];
  resolution_refs: string[];
  intent_shift: string | null;
};

export type BrainDumpResponse = {
  id: string;
  acknowledgement: string;
  signals: BrainDumpSignals;
  actionable: boolean;
};

export type BrainDumpRequest = {
  user_id: string;
  text: string;
};

export async function submitBrainDump(payload: BrainDumpRequest): Promise<{
  response: BrainDumpResponse;
  requestId: string | null;
}> {
  const { data, response } = await apiRequest<BrainDumpResponse>("/brain-dump", {
    method: "POST",
    body: payload,
  });

  return {
    response: data,
    requestId: response.headers.get("X-Request-Id"),
  };
}

File: backend/app/api/schemas/brain_dump.py
"""Pydantic schemas for brain dump API."""
from __future__ import annotations

from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, Field


class BrainDumpRequest(BaseModel):
    user_id: UUID
    text: str = Field(..., min_length=1, max_length=2000)


class BrainDumpSignals(BaseModel):
    emotional_state: Optional[str] = None
    blockers: List[str] = Field(default_factory=list)
    resolution_refs: List[str] = Field(default_factory=list)
    intent_shift: Optional[str] = None


class BrainDumpResponse(BaseModel):
    id: UUID
    acknowledgement: str
    signals: BrainDumpSignals
    actionable: bool
